debug/threads.py

```diff
@@ -167,25 +163,22 @@ def invoke(self, args, from_tty):
 
         # First try to search by thread_id
         try:
-            tid = int(args)
+            int(args)
         except ValueError:
             # If can't find by thread id, search by name
-            tds = []
-            for t in get_all_threads():
-                if thread_name(t) == args:
-                    tds.append(t)
-            if(len(tds) > 1):
-                print "Warning! There is more than 1 thread with name ", args
-            if(len(tds) > 0):
-                print(t)
+            tds = filter(lambda t: thread_name(t) == args, get_all_threads())
+            if len(tds) > 1:
+                print("Warning! There is more than 1 thread with name ", args)
+            if len(tds) > 0:
+                print(tds[0])
                 return
 
         for t in get_all_threads():
             if thread_id(t) == args:
                 print(t)
                 return
 
-        print "Can't find thread with '", args, "' id or name"
+        print("Can't find thread with '", args, "' id or name")
```

Comment:
Right! We need extra line `from __future__ import print_function` for Python 2.x!